src/resources/assets/js/components/settingsTabPlugin.vue

```diff
@@ -0,0 +1,32 @@
+<script>
+import MagicSamMixin from '../MagicSamMixin.vue';
+/**
+ * The plugin component to edit the Magic-SAM throttle interval.
+ *
+ * @type {Object}
+ */
+export default {
+    props: {
+        settings: {
+            type: Object,
+            required: true,
+        },
+    },
+    data() {
+        return {
+            restoreKeys: [
+                'SAMthrottleInterval',
+            ],
+            'SAMthrottleInterval':1000,
+        };
+    },
+    watch: {
+        SAMthrottleInterval(interval) {
+            interval = parseFloat(interval);
+            this.$emit('change', 'SAMthrottleInterval', interval);
+            this.settings.set('SAMthrottleInterval', interval);
+            MagicSamMixin.methods.set_throttleInterval(interval);
```

Comment:
Please use the global event bus instead of setting the value diretly on the object (which may cause problems with early or multiple instantiation). You can do it like [in the largo module](https://github.com/biigle/largo/blob/b8b1ae072fba8b2794710af0a1ae1c2c0cce62fd/src/resources/assets/js/components/annotationsSettingsTabPlugin.vue#L35-L39):

```js
watch: {
    isShown(shown) {
        Events.$emit('settings.exampleAnnotations', shown);
    },
},
created() {
    if (this.settings.has('exampleAnnotations')) {
        this.isShown = this.settings.get('exampleAnnotations');
    }
},
```

The bit in the `created()` hook should restore the initial value on page reload.

[And then](https://github.com/biigle/largo/blob/b8b1ae072fba8b2794710af0a1ae1c2c0cce62fd/src/resources/assets/js/components/annotationsLabelsTabPlugin.vue#L107-L109) (in MagicSamMixin):

```js
created() {
    Events.$on('settings.exampleAnnotations', this.updateShown);
},
```